### PR TITLE
feat: Add more type definitions

### PR DESCRIPTION
**Pull request by nzakas**
#### Prerequisites checklist

-   [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request?

Add some more low-hanging fruit types to `@eslint/core`

#### What changes did you make? (Give an overview)

Added several more types to `@eslint/core` that we can make use of now.

### CODE REVIEW

File: packages/core/src/types.ts [hunk 1]
```diff
@@ -59,12 +59,56 @@ export interface Position {
...
+	/**
+	 * Indicates whether or not inline configuration is evaluated.
+	 */
+	allowInlineConfig?: boolean;
+
+	/**
+	 * Indicates what to do when an unused disable directive is found.
+	 */
+	reportUnusedDisableDirectives?: boolean | SeverityName;
+}
```
Comment (mdjermanovic) [inline packages/core/src/types.ts:90]: These are options passed to `Linter#verify()` ([`VerifyOptions`](https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/linter/linter.js#L107-L118)), not `linterOptions` in config objects?

Comment (nzakas) [reply packages/core/src/types.ts:90]: No, this is for the config object. What am I missing?

Comment (mdjermanovic) [reply packages/core/src/types.ts:90]: In config objects (e.g., in an `eslint.config.js` file), [`linterOptions`](https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/config/flat-config-schema.js#L555-L559) objects have properties:

* `noInlineConfig`: allowed values are `true`, `false`.
* `reportUnusedDisableDirectives`: allowed values are `"error"`, `"warn"`, `"off"`, `2`, `1`, `0`, and also legacy `true`, `false`. https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/config/flat-config-schema.js#L287-L305

Comment (nzakas) [reply packages/core/src/types.ts:90]: Ah, so just missed the numbers. Thanks.

File: packages/core/src/types.ts [hunk 2]
```diff
@@ -59,12 +59,56 @@ export interface Position {
...
+export type Severity = SeverityName | SeverityLevel;
+
+/**
+ * Represents the configuration options for the core linter.
+ */
+export interface LinterOptionsConfig {
+	/**
+	 * Indicates whether or not inline configuration is evaluated.
+	 */
+	allowInlineConfig?: boolean;
```
Comment (mdjermanovic) [inline packages/core/src/types.ts:89]:
```suggestion
	noInlineConfig?: boolean;
```

The config option is `noInlineConfig`.

https://github.com/eslint/eslint/blob/7bcda760369c44d0f1131fccaaf1ccfed5af85f1/lib/config/flat-config-schema.js#L555-L559

**Review by mdjermanovic — APPROVED** [review]
LGTM, thanks!

**Pull request merged**